### PR TITLE
table: Add arrow-compatible types

### PR DESCRIPTION
**Pull request by ibgreen**

### CODE REVIEW

File: modules/shapefile/src/lib/parsers/parse-dbf.js [hunk 1]
```diff
@@ -50,10 +52,20 @@ export function parseDBF(arrayBuffer, options) {
...
-  return data;
-}
+  // const data = parseRows(binaryReader, fields, nRecords, recordLength, textDecoder);
+
+  const {data, schema} = dbfParser.result;
 
+  switch (options.tables && options.tables.format) {
+    case 'table':
+      // TODO - parse columns
+      return {schema, rows: data};
```
Comment (kylebarron) [inline modules/shapefile/src/lib/parsers/parse-dbf.js:62]: So this doesn't return `rows` _as an Arrow table_, it returns as a normal JS object with the Arrow metadata of what each key should be understood as?

File: modules/shapefile/src/lib/parsers/parse-dbf.js [hunk 2]
```diff
@@ -251,3 +264,29 @@ function parseNumber(text) {
 function parseCharacter(text) {
   return text.trim() || null;
 }
+
+/**
+ * Create a standard Arrow-style `Field` from field descriptor.
+ * TODO - use `fieldLength` and `decimal` to generate smaller types?
+ * TODO - is everything nullable. Can we set this false for some fields?
```
Comment (kylebarron) [inline modules/shapefile/src/lib/parsers/parse-dbf.js:271]: If I recall correctly that isn't in the DBF's metadata, so every column needs to be considered nullable.

File: modules/tables/src/lib/schema/field.js [hunk 3]
```diff
@@ -0,0 +1,45 @@
+// Copyright (c) 2020 Unfolded, Inc. All rights reserved.
```
Comment (kylebarron) [inline modules/tables/src/lib/schema/field.js:1]: Is this copyright statement intentional here?

File: modules/tables/src/lib/schema/field.js [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+import assert from '../utils/assert';
+import {DataType} from '../types/arrow-like/type';
+
+// ArrowJS `Schema` API-compatible class for row-based tables (returned from `DataTable`)
```
Comment (kylebarron) [inline modules/tables/src/lib/schema/field.js:5]: I see, so you don't depend on the Arrow JS library directly within the `tables` module. Makes a lot of sense; presumably it's easy to translate these fields into actual Arrow fields.

File: modules/shapefile/src/lib/parsers/parse-dbf.js (hunk 1 above)
Comment (ibgreen) [reply modules/shapefile/src/lib/parsers/parse-dbf.js:62]: Yes - converting tables and batches to Arrow format is a bigger thing. 

Here I just want to create a schema that describes the binary format of the columns. This schema will obviously be used when actually converting the output to Arrow.

File: modules/tables/src/lib/schema/field.js (hunk 4 above)
Comment (ibgreen) [reply modules/tables/src/lib/schema/field.js:5]: Yes ArrowJS is still a bit too big a dependency for comfort, but my hope is eventually we can remove these helper classes. 

Until then we can trivially convert as you say.

**Pull request merged**